build_tools/linux_python_package.py

```diff
@@ -0,0 +1,192 @@
+  a relative symlink (also rewriting shared library soname links as needed).
+  Since symlinks and non-standard attributes cannot be included in a wheel file,
+  the platform contents are stored in a `_devel.tar` or `_devel.tar.xz` file.
+  The installed package is extended in response to requesting a path to it
+  via the `rocm-sdk` tool.
+
+Runtime packages can either be target neutral or target specific. Target specific
+packages are suffixed with their target family and the setup files have special logic
+to determine what is correct to load based on the system. In the future, this
+will be re-arranged to have more of a thin structure where the host code is
+always emitted as target neutral and separate device code packages are loaded
+as needed.
+
+It is expected that all packages are installed in the same site-lib, as they use
+relative symlinks and RPATHs that cross the top-level package boundary. The
+built-in tests (via `rocm-sdk test`) verify these conditions.
+"""
+
+import argparse
+import functools
+import re
+from pathlib import Path
+import sys
+
+from _therock_utils.artifacts import ArtifactCatalog, ArtifactName
+from _therock_utils.py_packaging import Parameters, PopulatedDistPackage, build_packages
+
+
+def run(args: argparse.Namespace):
+    params = Parameters(
+        dest_dir=args.dest_dir,
+        version=args.version,
+        version_suffix=args.version_suffix,
+        artifacts=ArtifactCatalog(args.artifact_dir),
+    )
+
+    # Simple populate the top-level "rocm-sdk" package. This gets no platform files.
+    PopulatedDistPackage(params, logical_name="meta")
+
+    # Populate each target neutral library package.
+    core = PopulatedDistPackage(params, logical_name="core").populate_runtime_files(
+        params.filter_artifacts(
+            core_artifact_filter,
+            # TODO: The base package is shoving CMake redirects into lib.
+            excludes=["**/cmake/**"],
+        ),
+    )
+
+    # Populate each target-specific library package.
+    for target_family in params.all_target_families:
+        lib = PopulatedDistPackage(
+            params, logical_name="libraries", target_family=target_family
+        )
+        lib.rpath_dep(core, "lib")
+        lib.rpath_dep(core, "lib/rocm_sysdeps/lib")
+        lib.rpath_dep(core, "lib/host-math/lib")
+        lib.populate_runtime_files(
+            params.filter_artifacts(
+                filter=functools.partial(libraries_artifact_filter, target_family),
+            )
+        )
+
+    # And populate the devel package, which catches everything else.
+    devel = PopulatedDistPackage(params, logical_name="devel")
+    devel.populate_devel_files(tarball_compression=args.devel_tarball_compression)
+
+    if args.build_packages:
+        build_packages(args.dest_dir, wheel_compression=args.wheel_compression)
+
+
+def core_artifact_filter(an: ArtifactName) -> bool:
+    core = an.name in [
```

Comment:
Is `core` equivalent to `generic` here _or_ does it rather specify core components that should be always required? If it's the latter, `host-blas` could be moved to libraries as it _should_ be only needed for the blas artifact. It was also organized this way before thus some clarification would be nice.

Reply:
Yes, how it was before was an oversight: we generate one libraries package per target, so it wasn't the best place to have a shared host side dep. It was also broken where it was because it is a generic package and not being included in the target specific filter. 

Now that I have this less hard coded, I'm more free to relayout the packages. That will yield a better solution here.